tool/tsh/common/workload_identity.go

```diff
@@ -0,0 +1,219 @@
+/*
+ * Teleport
+ * Copyright (C) 2024  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package common
+
+import (
+	"bytes"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"os"
+	"path"
+	"time"
+
+	"github.com/alecthomas/kingpin/v2"
+	"github.com/gravitational/trace"
+	"google.golang.org/protobuf/types/known/durationpb"
+
+	"github.com/gravitational/teleport"
+	machineidv1pb "github.com/gravitational/teleport/api/gen/proto/go/teleport/machineid/v1"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/auth/native"
+	"github.com/gravitational/teleport/lib/client"
+	"github.com/gravitational/teleport/lib/services"
+)
+
+type workloadIdentityCommands struct {
+	issue *workloadIdentityIssueCommand
+}
+
+func newWorkloadIdentityCommands(app *kingpin.Application) workloadIdentityCommands {
+	cmd := app.Command("workloadid", "Manage Teleport Workload Identity.")
+	cmds := workloadIdentityCommands{
+		issue: newWorkloadIdentityIssueCommand(cmd),
+	}
+	return cmds
+}
+
+const (
+	// Based on the default paths listed in
+	// https://github.com/spiffe/spiffe-helper/blob/main/README.md
+	svidPEMPath            = "svid.pem"
+	svidKeyPEMPath         = "svid_key.pem"
+	svidTrustBundlePEMPath = "svid_bundle.pem"
+
+	svidTypeX509 = "x509"
+)
+
+type workloadIdentityIssueCommand struct {
+	*kingpin.CmdClause
+	svidPath        string
+	svidType        string
+	svidDNSSANs     []string
+	svidIPSANs      []string
+	svidTTL         time.Duration
+	outputDirectory string
+}
+
+func newWorkloadIdentityIssueCommand(parent *kingpin.CmdClause) *workloadIdentityIssueCommand {
+	cmd := &workloadIdentityIssueCommand{
+		CmdClause: parent.Command("issue", "Issues a SPIFFE SVID using Teleport Workload Identity and writes it to a local directory."),
+	}
+	cmd.Arg("path", "Path use for the SVID SPIFFE ID. Must have a preceding '/'.").
+		Required().
+		StringVar(&cmd.svidPath)
```

Comment:
Nit, but shouldn't the struct field be called `spiffeIDPath` in this case?

Also:

```suggestion
	cmd.Arg("path", "Path to use for the SVID SPIFFE ID. Must have a preceding '/'.").
		Required().
		StringVar(&cmd.svidPath)
```

Reply:
4df6d3605f